zigzag/stages/SearchUnusedMemoryStage.py

```diff
@@ -156,14 +150,14 @@ def update_top_mem_level(self):
             output_operand_in_layer: LayerOperand = layer.output_operand
             output_operand_in_hardware: MemoryOperand = layer.memory_operand_links[output_operand_in_layer]
 
-            is_branch_starting_node = True if self.workload.out_degree(layer) > 1 else False
-            is_branch_final_node: bool = (self.workload.out_degree(layer) == 1) and (
```

Comment:
@JiacongSun  This was also set to `out_degree` before the refactoring, so this cannot be the cause of the failing tests

Reply:
I mean the 2nd part of the condition for `is_branch_final_node`